nudge/src/utils/keyData.ts

```diff
@@ -0,0 +1,54 @@
+import { ethers } from 'ethers'
+
+export default class KeyData {
+  provider: ethers.providers.JsonRpcProvider
+
+  constructor(provider: string) {
```

Comment:
```suggestion
  constructor(providerUrl: string) {
```

Minor suggestion, just to remove ambiguity for future reading/modification